dali/kernels/imgproc/resample/resampling_setup.cc

```diff
@@ -21,19 +21,75 @@ namespace kernels {
 inline ResamplingFilter GetResamplingFilter(
     const ResamplingFilters *filters, const FilterDesc &params) noexcept {
   switch (params.type) {
-    case ResamplingFilterType::Gaussian:
-      return filters->Gaussian(params.radius*0.5f/M_SQRT2);
+    case ResamplingFilterType::Linear:
+      return filters->Triangular(1);
     case ResamplingFilterType::Triangular:
       return filters->Triangular(params.radius);
-    case ResamplingFilterType::Lanczos3:
-      return filters->Lanczos3();
+    case ResamplingFilterType::Gaussian:
+      return filters->Gaussian(params.radius*0.5f/M_SQRT2);
     case ResamplingFilterType::Cubic:
       return filters->Cubic();
+    case ResamplingFilterType::Lanczos3:
+      return filters->Lanczos3();
     default:
       return { nullptr, 0, 0, 0 };
   }
 }
 
+void SeparableResamplingSetup::SetFilters(SampleDesc &desc, const ResamplingParams2D &params) {
+  const auto &out_shape = desc.shapes[IdxOut];
+  for (int axis = 0; axis < 2; axis++) {
+    float in_size;
+    if (params[axis].roi.use_roi) {
+      in_size = std::abs(params[axis].roi.end - params[axis].roi.start);
+    } else {
+      in_size = desc.shapes[IdxIn][axis];
+    }
+
+    auto fdesc = out_shape[axis] < in_size ? params[axis].min_filter
+                                           : params[axis].mag_filter;
+    if (fdesc.radius == 0)
+      fdesc.radius = DefaultFilterRadius(fdesc.type, in_size, out_shape[axis]);
+    desc.filter_type[axis] = fdesc.type;
+    auto &filter = desc.filter[axis];
+    filter = GetResamplingFilter(filters.get(), fdesc);
+  }
+}
+
+SeparableResamplingSetup::ROI SeparableResamplingSetup::ComputeScaleAndROI(
+    SampleDesc &desc, const ResamplingParams2D &params) {
```

Comment:
I really hate to be thins kind of guy that talks all the time about one thing only, but this function asks for unit tests...

Reply:
I don't like the way this function turned out, to be honest. I depends on some intermediate state of `SampleDesc`, it depends on filters' properties... It's hard to unit-test and actually I made it protected in the next PR - and maybe should even make it private, as it makes no sense to use it outside the computation setup.